src/osd/scrubber/pg_scrubber.cc

```diff
@@ -1113,10 +1038,8 @@ int PgScrubber::build_replica_map_chunk()
 
     case 0: {
       // finished!
-      m_cleaned_meta_map.clear_from(m_start);
-      m_cleaned_meta_map.insert(replica_scrubmap);
-      auto for_meta_scrub = clean_meta_map();
-      _scan_snaps(for_meta_scrub);
+
+      m_be->replica_clean_meta(replica_scrubmap, m_end.is_max(), m_start);
```

Comment:
This should probably be part of build_scrub_map_chunk()?

Reply:
That will be a change in order of operations for the Primary. Maybe can be done - but surely not for this PR.